python/ray/train/utils.py

```diff
@@ -172,3 +173,88 @@ def __getattr__(self, item):
         # actor.
         actor_method = getattr(self.actor, item)
         return lambda *args, **kwargs: ray.get(actor_method.remote(*args, **kwargs))
+
+
+@dataclass
+class _RayDatasetSpec:
+    """Configuration for Ray Datasets to pass to the training workers.
+
+    dataset_or_dict: An optional Ray Dataset (or DatasetPipeline) or a dictionary of
+        datasets to be sharded across all the training workers, which can be accessed
+        from the training function via ``train.get_dataset_shard()``. Multiple Datasets
+        can be passed in as a ``Dict`` that maps each name key to a Dataset value,
```

Comment:
Super-nit: should it be ``Dict`` or ``dict``? Python 3.10+ ``Dict`` is discouraged.

Reply:
It seems that Dict is needed if we want to also provide type hints for the keys and values. In <3.9, it's not possible to do this with dict.

Reply:
It should be `dict` in the docstring though, as that is the actual class and not the type hint.

Reply:
The docstring has been updated.

Reply:
Awesome, had to refresh :)